### PR TITLE
Misc: updated .gitignore

### PR DESCRIPTION
**Pull request by midenok**
Please, merge.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,5 @@
 *-t
 *.ctest
-*.reject
```
Comment (svoj) [inline .gitignore]: Why is that?

Comment (midenok) [reply .gitignore]: Because it is junk and should be deleted by `git clean -f`.

Comment (svoj) [reply .gitignore]: Well, from this point of view ```*.o``` are probably junk as well. As well as other generated files that are in ```.gitignore```.

I'm not completely sure here. So I'd better bring this up to other devs.

Comment (midenok) [reply .gitignore]: Yes, there are some trash that is not supposed to be here (like `*.tmp` f.ex.). OTOH there are no `*.orig` which comes as a pair to `*.reject`.

Comment (svoj) [reply .gitignore]: Ok, so there're two kinds of ```*.reject```: those that are created by patch and those that are generated by mysql-test-run.pl. Still, it needs some thinking.

**Comment by vuvova** [comment]
There's no "junk" category in git, files are either ignored by `git comit` or aren't. So everything that shouldn't be committed, should be in `.gitignore`

File: .gitignore (hunk 1 above)
Comment (midenok) [reply .gitignore]: Just figured out that those that come from `patch` are in fact `*.rej`. I have nothing against `*.reject`. Updated commit.

**Comment by midenok** [comment]
"Junk" category is a human term who uses `git` for his purposes. So this tool should be configured to produce optimal response.

**Comment by svoj** [comment]
Updated patch looks good. Let's see if anybody still wants to object.

**Comment by svoj** [comment]
Thanks!

**Pull request merged**